### DISCUSSION TITLE
Chrysler Pacifica Hybrid  PID Expression Help

### DISCUSSION THREAD

**Discussion by smurf12345** [meatpiHQ/wican-fw#254] [Q&A] [answered]
Hi All:  
I need some pointing in the right direction.  I just converted my old node-red mqtt requests to the new AutoPid Vehicle Profile.  I have Coolant, Fuel Level, Odometer, Hybrid Battery Pack Remaining, and Ambient Temperature working.  However, I'm having a hard time with  Battery Cooling Inlet Temp, Battery Cooling Outlet Temp, BPCM Module Temp Sensors (1-12), Warm-Ups Since Code Cleared, and Distance to Empty.  I think I am not understanding the logic for multi-line messages or something else.  Any pointers would be much appreciated as I've gone in circles on this one.

Here is what I have so far: [vehicle_profiles-ChryslerPacificaHybrid](https://github.com/smurf12345/home-assistant/blob/main/vehicle_profiles-ChryslerPacificaHybrid.json)

For example, BPCM Module Temp Sensors 1-12 appear to be 1 requested PID, but I must not be writing the expression correctly as it shows Unavailable in Home-assistant.

ELM327 Request/Response:  Sensor 1 and Sensor 12 I believe are reporting 20C while Sensors 2-11 reporting 21C based on what CarScanner is showing me.  What am I missing here?

```
>22A0664
7EF101B62A0663C3D3D
7EF213D3D3D3D3D3D3C
7EF223D3C2828282828
7EF2328282828282828

       {
          "pid": "22A0664",
          "parameters": [
            {
              "name": "BPCM Module Temp Sensor 1",
              "expression": "((B3 - 174) * 1.8) + 32",
              "unit": "°F",
              "class": "temperature"
            },
            {
              "name": "BPCM Module Temp Sensor 2",
              "expression": "((B4 - 190) * 1.8) + 32",
              "unit": "°F",
              "class": "temperature"
            }, // Etc... repeating until sensor 12
       }
```

I'm also having problems with Warm-Ups Since Code Cleared and Distance Since Codes Cleared Miles.  I don't think I'm filtering correctly as multiple modules port back.  I did a pid_init: "ATCRA7EC" to just get that module.  I sometimes get the correct numbers which match what CarScanner is reporting, but then it seems the numbers will incorrectly vary (car hasn't moved so they should be static).

For example: Distance traveled since codes cleared = 5202.12 miles (8,371.97 km)
```
>0131
7E804413120B4
7EC04413120C8
7EA0441312086
7EB0441310000
7EF04413120C8

        {
            "pid": "0131",
            "parameters": [
              {
                "name": "Distance Since Codes Cleared Miles",
                "pid_init": "ATCRA7EC;",
                "expression": "(B3 * 256 + B4) * 0.621371",
                "unit": "miles",
                "class": "distance"
              }
            ]
          }
```

Same for Warm-ups Since Code Cleared: It should report 65, but it will vary from 0, 65, 230, 231.
```
>0130 
7E8 03 41 30 41
7EA 03 41 30 E6
7EC 03 41 30 E7
7EB 03 41 30 00
7EF 03 41 30 E7

        {
          "pid": "0130",
          "parameters": [
            {
              "name": "Warm-Ups Since Code Cleared",
              "pid_init": "AT SH 7E0; AT CRA 7E8;",
              "expression": "B3",
              "unit": "count",
              "class": "diagnostic"
            }
          ]
        },
```

**Comment by meatpiHQ** [comment, marked as answer]
@smurf12345 lets try to figure these out one by one. To find the pid_init you should look at car scanner logs, please see the link below, the documentation is not yet complete be just to give you an idea. You can post the car scanner logs here and I can try to help.

https://meatpihq.github.io/wican-fw/config/automate/new_vehicle_profiles

**Reply by smurf12345** [reply]
Thank you!  That got me going.  It was the init and pid_init that I completely missed.  After following the docs above and monitoring the logs, I saw the what init values it was using.  They are now reporting values correctly.  Thanks again!

**Reply by meatpiHQ** [reply]
@smurf12345 If it's working for you can please make a pull request with your profile?
